### ISSUE TITLE
[Bug]: "import matplotlib; print(matplotlib.__version__)" fails on PyPy-3.8v7.3.8rc1

### ISSUE THREAD

**Issue by stonebig** [matplotlib/matplotlib#22357]
### Bug summary

I install PyPy-3.8v7.3.8rc1, and then matplotlib-3.5.1 (or 3.5.0 or 3.4.3) wheel from cgohlke.

I start IDLE and launch

import matplotlib.pyplot as plt

### Code for reproduction

```python
import matplotlib.pyplot as plt
```

### Actual outcome

```
Python 3.8.12 (67f1b98040bad2f1241e4e1ebde6051a505f628a, Jan 26 2022, 20:56:38)
[PyPy 7.3.8 with MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> import matplotlib.pyplot as plt
Traceback (most recent call last):
  File "<pyshell#0>", line 1, in <module>
    import matplotlib.pyplot as plt
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\__init__.py", line 1452, in <module>
    from matplotlib.cm import _colormaps as colormaps
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\cm.py", line 187, in <module>
    _cmap_registry = _gen_cmap_registry()
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\cm.py", line 51, in _gen_cmap_registry
    colors.LinearSegmentedColormap.from_list(name, spec, _LUTSIZE))
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\colors.py", line 954, in from_list
    vals = np.linspace(0, 1, len(colors))
  File "<__array_function__ internals>", line 6, in linspace
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\numpy\core\function_base.py", line 130, in linspace
    dt = result_type(start, stop, float(num))
  File "<__array_function__ internals>", line 6, in result_type
TypeError: Cannot interpret '9.0' as a data type
>>> 
````
```

### Expected outcome

no error

### Additional information

_No response_

### Operating system

Windows 10

### Matplotlib Version

3.4.3 (but 3.5.0 and 3.5.1 too)

### Matplotlib Backend

TkAgg

### Python version

Python 3.8.12 (67f1b98040bad2f1241e4e1ebde6051a505f628a, Jan 26 2022, 20:56:38) [PyPy 7.3.8 with MSC v.1929 64 bit (AMD64)]

### Jupyter version

_No response_

### Installation

pip

**Comment by stonebig** [comment]
even worth with matplotlib-3.5.1, the error is on  `import matplotlib; print(matplotlib.__version__)`

```
Python 3.8.12 (67f1b98040bad2f1241e4e1ebde6051a505f628a, Jan 26 2022, 20:56:38)
[PyPy 7.3.8 with MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> import matplotlib; print(matplotlib.__version__)
Traceback (most recent call last):
  File "<pyshell#0>", line 1, in <module>
    import matplotlib; print(matplotlib.__version__)
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\__init__.py", line 1452, in <module>
    from matplotlib.cm import _colormaps as colormaps
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\cm.py", line 187, in <module>
    _cmap_registry = _gen_cmap_registry()
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\cm.py", line 51, in _gen_cmap_registry
    colors.LinearSegmentedColormap.from_list(name, spec, _LUTSIZE))
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\matplotlib\colors.py", line 954, in from_list
    vals = np.linspace(0, 1, len(colors))
  File "<__array_function__ internals>", line 6, in linspace
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\numpy\core\function_base.py", line 130, in linspace
    dt = result_type(start, stop, float(num))
  File "<__array_function__ internals>", line 6, in result_type
TypeError: Cannot interpret '9.0' as a data type
>>> 
```

**Comment by timhoffm** [comment]
Does
```
import numpy as np
np.linspace(0, 1, 9)
```
Result in the same error?

**Comment by stonebig** [comment]
yes , so it's a numpy or numpy wheel problem

````
Python 3.8.12 (67f1b98040bad2f1241e4e1ebde6051a505f628a, Jan 26 2022, 20:56:38)
[PyPy 7.3.8 with MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> import numpy as np;np.linspace(0, 1, 9)
Traceback (most recent call last):
  File "<pyshell#0>", line 1, in <module>
    import numpy as np;np.linspace(0, 1, 9)
  File "<__array_function__ internals>", line 6, in linspace
  File "C:\WinP\bd38\budotPyPy\WPy64-38121b6\pypy3.8-v7.3.8rc1-win64\lib\site-packages\numpy\core\function_base.py", line 130, in linspace
    dt = result_type(start, stop, float(num))
  File "<__array_function__ internals>", line 6, in result_type
TypeError: Cannot interpret '9.0' as a data type
>>> 
````

**Comment by stonebig** [comment]
well, the problem is why this is called 

![image](https://user-images.githubusercontent.com/4312421/151693479-1d48810e-36f1-41a9-8685-1f7a65e49721.png)

**Comment by stonebig** [comment]
np.linspace(0, 1, **len(colors)**) is the problem I suppose

**Comment by timhoffm** [comment]
Yes, but `linspace(start, stop, num)` is the correct API. https://numpy.org/doc/stable/reference/generated/numpy.linspace.html

So I don‘t think the issue is on our side.

**Comment by stonebig** [comment]
yes, you're right. sorry for the re-opening

**Comment by stonebig** [comment]
solved in PyPy-3.8 branch, thanks for the hints towards the root problem
